src/components/common/CommentEditor.js

```diff
@@ -0,0 +1,244 @@
+    } else {
+      // console.log('calling handle edit clicked')
+      onEditComment(comment.id);
+    }
+    this.setState({ isEdited: !isEdited });
+  };
+
+  onComDelete = (id) => {
+    const { onDeleteComment } = this.props;
+    // console.log('comment to be deleted', id);
+    onDeleteComment(id);
+  };
+
+  renderCardHeader() {
+    /* eslint-disable react/prop-types */
+    const { comment, classes } = this.props;
+    const { author, date } = comment;
+    const { isEdited, showDelete, open } = this.state;
+    return isEdited ? null : (
+      <CardHeader
+        className={classes.header}
+        avatar={
+          <Avatar>
+            {author
+              .match(/\b(\w)/g)
+              .slice(0, 2)
+              .join('')}
+          </Avatar>
+        }
+        title={author}
+        subheader={date}
+        action={
+          <>
+            {showDelete ? (
+              <>
+                <IconButton
+                  aria-label="edit"
+                  color="primary"
+                  onClick={this.onEdit}
+                >
+                  <EditIcon />
+                </IconButton>
+                <IconButton
+                  aria-label="delete"
+                  color="secondary"
+                  onClick={() => this.setState({ open: true })}
+                >
+                  <DeleteIcon />
+                </IconButton>
+              </>
+            ) : null}
+            <ConfirmDialog
+              open={open}
+              setOpen={(v) => this.setState({ open: v })}
+              onClose={(m) => {
+                if (m) {
+                  this.onComDelete(comment.id);
+                }
+              }}
+            />
+          </>
+        }
+      />
+    );
+  }
+
+  render() {
+    const { selectedTab, value, isEdited } = this.state;
+    const { classes } = this.props;
+    return (
+      <Card
+        className={classes.root}
+        onMouseEnter={() => this.setState({ showDelete: true })}
+        onMouseLeave={() => this.setState({ showDelete: false })}
+        variant="outlined"
+      >
+        {this.renderCardHeader()}
+        <CardContent className={classes.content}>
+          {isEdited ? (
+            <ReactMde
+              value={value}
+              onChange={(v) => this.setState({ value: v })}
+              selectedTab={selectedTab}
+              onTabChange={(t) => this.setState({ selectedTab: t })}
+              generateMarkdownPreview={(markdown) =>
+                Promise.resolve(this.converter.makeHtml(markdown))
+              }
+              childProps={{
+                writeButton: {
+                  tabIndex: -1,
+                },
+                textArea: {
+                  autoFocus: true,
+                },
+              }}
+            />
+          ) : (
+            <Grid
+              container
+              className="mde-preview standalone"
+              style={{ paddingLeft: '38px' }}
```

Comment:
Try to put styles using `styles`.

Reply:
Yes i had my fair shar of strugling because it did not want to work until i found a way of making it 😅